cmake/packaging.cmake

```diff
@@ -197,20 +212,62 @@ function(generateInstallTargets)
     # .
     install(PROGRAMS "$<TARGET_FILE:osqueryd>" DESTINATION . RENAME osqueryi.exe)
 
-    file(COPY "${CMAKE_SOURCE_DIR}/tools/deployment/osquery.example.conf" DESTINATION "${CMAKE_BINARY_DIR}/package/wix")
-    install(FILES "${CMAKE_BINARY_DIR}/package/wix/osquery.example.conf" DESTINATION . RENAME osquery.conf)
+    if("${PACKAGING_SYSTEM}" STREQUAL "NuGet")
+      set(win_packager "nuget")
+      # Grab the Chocolatey management scripts
+      file(COPY "${CMAKE_SOURCE_DIR}/tools/deployment/chocolatey/tools" DESTINATION "${CMAKE_BINARY_DIR}/package/${win_packager}")
+      install(DIRECTORY "${CMAKE_BINARY_DIR}/package/${win_packager}/tools" DESTINATION .)
+
+      # Chocolatey requires a LICENSE.txt and VERIFICATION.txt file in all 
+      # packages containing .exe, .msi, or .zip files.
+      file(COPY "${CMAKE_SOURCE_DIR}/LICENSE" DESTINATION "${CMAKE_BINARY_DIR}/package/${win_packager}")
+      install(FILES "${CMAKE_BINARY_DIR}/package/${win_packager}/LICENSE" DESTINATION . RENAME LICENSE.txt)
+
+      file(WRITE "${CMAKE_BINARY_DIR}/package/${win_packager}/VERFICATION.txt" "
+      VERIFICATION
+
+      Verification is intended to assist the Chocolatey moderators and community
+      in verifying that this package's contents are trustworthy.
+      
+      Package can be verified like this:
+
+        - Check that the file signing certificate is valid:
+        ```
+        PS C:\\Users\\thor> Get-AuthenticodeSignature 'C:\\Program Files\\osquery\\osqueryd\\osqueryd.exe'
+          Directory: C:\\Program Files\\osquery\\osqueryd
+
+          SignerCertificate                         Status          Path
+          -----------------                         ------          ----
+          EEA3330A4F0FAC86A3FE01F1E7D72E7A9808AC86  Valid           osqueryd.exe
+        ```
+
+        - Check that the file hash of the osqueryd.exe binary is correct:
+        ```
+        PS C:\\Users\\thor> (Get-FileHash -Algorithm sha256 'C:\\Program Files\\osquery\\osqueryd\\osqueryd.exe').Hash
+        <Fill in with osqueryd.exe Sha256 before publishing to Chocolatey>
```

Comment:
I don't really like doing it like this, but I'm not sure howe to get the file digest of the osqueryd.exe binary before `cpack` is invoked. Right now we already need to manually swap out the binaries with the signed versions before calling `cpack` anyway, so I'm going to say we just manually update the `verification.txt`, unless someone smarter than I in cmake can tell me how to get the sha256 of osqueryd.exe at this point :)

Reply:
You can add a custom command that run after the build of a target, check this https://cmake.org/cmake/help/v3.16/command/add_custom_command.html and POST_BUILD.